blog/2023-11-01-feature-flags-sdks-architectures.md

```diff
@@ -0,0 +1,127 @@
+
+### "Direct" API endpoints Bridge SDK - example flow
+
+<img src={require('@site/static/img/blog/feature-flags-sdks-architectures/Direct-API-endpoints-Bridge-SDK.png').default} />
+
+### "Direct" API endpoints Bridge SDK using Relay Proxy - example flow
+
+<img src={require('@site/static/img/blog/feature-flags-sdks-architectures/Direct-API-endpoints-Bridge-SDK-using-Relay-Proxy.png').default} />
+
+#### Advantages
+
+* Can be relatively simple to develop and maintain from the APIs in multiple code languages, even by code generation.
+  The evaluation code can reside at the Relay Proxy, without a need to implement evaluation per SDK language.
+
+#### Disadvantages
+
+* Functional disadvantage is that intermediate issue like network or feature flags service error is handled by not evaluating to the expected feature flag value.
+* Another functional disadvantage is that intermediate issue like network or feature flags service error during microservice initialization is handled by not evaluating to the expected feature flag value.
+* Performance and resources usage: each operation like feature flag evaluation causing network traffic.
+    * Time-consuming.
+    * High load on network traffic.
+    * Affecting both application, and Feature Flag service itself, whether it is self-hosted relay proxy or even more when it is a Cloud Saas service.
+
+### API endpoints requests with cache
+
+With this approach, the feature flag service exposes APIs via endpoints like REST HTTP and/or gRpc, and the SDK calling the endpoint requests, and translate the request/responses to the programming language. A cache is used for responses data.  
+Feature flag evaluation is done on proxy / features service only.  
+As this approach involves network traffic for non-cached evaluations, a Relay Proxy is effective for having the network traffic data on internal network for faster and efficient performance and reducing the load on the Features Service / Source.
+
+### API endpoints requests with cache - example flow
+
+<img src={require('@site/static/img/blog/feature-flags-sdks-architectures/API-endpoints-requests-with-cache.png').default} />
+
+### API endpoints requests with cache using Relay Proxy - example flow
+
+<img src={require('@site/static/img/blog/feature-flags-sdks-architectures/API-endpoints-requests-with-cache-using-Relay-Proxy.png').default} />
+
+#### Advantages
+
+* Network traffic can be reduced.
+
+#### Disadvantages
+
+* Functional disadvantage is that intermediate issue like network or feature flags service error for non-cached data is handled by not evaluating to the expected feature flag value.
+* Another functional disadvantage is that intermediate issue like network or feature flags service error during microservice initialization is handled by not evaluating to the expected feature flag value.
```

Comment:
Please see my comments in the `"Direct" API endpoints bridge` section. We should keep these points in line.